activerecord/lib/active_record/connection_adapters/mysql/database_statements.rb

```diff
@@ -11,7 +11,10 @@ module DatabaseStatements
 
         # https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_current-timestamp
         # https://dev.mysql.com/doc/refman/5.7/en/date-and-time-type-syntax.html
-        HIGH_PRECISION_CURRENT_TIMESTAMP = Arel.sql("CURRENT_TIMESTAMP(6)", retryable: true).freeze # :nodoc:
+        HIGH_PRECISION_CURRENT_TIMESTAMP = case ActiveRecord.default_timezone # :nodoc:
+        when :utc   then Arel.sql("UTC_TIMESTAMP(6)", retryable: true).freeze
+        when :local then Arel.sql("CURRENT_TIMESTAMP(6)", retryable: true).freeze
+        end
```

Comment:
If the default timezone is set by Rails does this change at the right time? It might make more sense to keep both as constants and switch when the method is invoked below: `#high_precision_current_timestamp`. Seems easiler to test too.

Reply:
Yeah, I wasn't sure about this...

I think we want to reduce processing as much as possible when calling `insert_all` (think bulk imports of huge CSVs, any small thing could have an impact), so I wanted to avoid to use an `if/else` for every insertion.

`ActiveRecord.default_timezone` is usually set at boot so I assumed we'd be fine, though now I understand that this would likely be loaded before the app loads and gets a chance to set it...

Reply:
https://github.com/rails/rails/blob/299b6510348606380c21cc6e599c99409f6ec19a/activerecord/lib/active_record/insert_all.rb#L222

Looking at this, I guess we'd be calling `high_precision_current_timestamp` only once per `insert_all`, so an `if/else` would be viable). Will update.

Reply:
@gmcgibbon I updated my code.